File: study/lua/src/app/scene/view/mineCraft/MineMoveAvatar.lua
local ViewBase = require("app.ui.ViewBase")
local MineMoveAvatar = class("MineMoveAvatar", ViewBase)

-- CampRacePreMatch.testTime = 60

function MineMoveAvatar:ctor()
	local resource = {
		file = Path.getCSB("MineMoveAvatar", "mineCraft"),
		size = G_ResolutionManager:getDesignSize(),
		binding = {

		}
	}
    MineMoveAvatar.super.ctor(self, resource)
end

function MineMoveAvatar:onCreate()
end

function MineMoveAvatar:onEnter()
end

function MineMoveAvatar:onExit()
end

function MineMoveAvatar:updateUI(heroId, name, officel, limit,titleId)
    self._heroAvatar:updateUI(heroId, nil, nil, limit)
    self._textName:setString(name)
    self._textName:setColor(Colors.getOfficialColor(officel))
    self._textName:enableOutline(Colors.getOfficialColorOutline(officel), 2)
    local officalInfo, officalLevel = G_UserData:getBase():getOfficialInfo(officel)
    if officalLevel <= 0 then --无不显示
        self._textOfficial:setVisible(false)
		return
    end
	self._textOfficial:setColor(Colors.getOfficialColor(officalLevel))
	self._textOfficial:enableOutline(Colors.getOfficialColorOutline(officalLevel), 2)
	self._textOfficial:setVisible(true)
    self._textOfficial:setString("["..officalInfo.name.."]")
    self._heroAvatar:showTitle(titleId,self.__cname)
end

function MineMoveAvatar:setAction(action, loop)
    self._heroAvatar:setAction(action, loop)
end

function MineMoveAvatar:turnBack(needTurn)
    self._heroAvatar:turnBack(needTurn)
end

return MineMoveAvatar

File: study/ts/script/scene/view/mineCraft/MineMoveAvatar.ts
const { ccclass, property } = cc._decorator;

import CommonHeroAvatar from '../../../ui/component/CommonHeroAvatar'
import { Colors, G_UserData } from '../../../init';
import UIHelper from '../../../utils/UIHelper';
import { BaseData } from '../../../data/BaseData';
import { UserBaseData } from '../../../data/UserBaseData';
import ViewBase from '../../ViewBase';

@ccclass
export default class MineMoveAvatar extends ViewBase {

    @property({
        type: CommonHeroAvatar,
        visible: true
    })
    _heroAvatar: CommonHeroAvatar = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textName: cc.Label = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textOfficial: cc.Label = null;
    constructor() {
        super();
    }

    onCreate() {
        this.node.name = "MineMoveAvatar";
        this._heroAvatar.init();
    }
    onEnter() {
    }
    onExit() {
    }
    updateUI(heroId, name, officel, limit, titleId) {
        this._heroAvatar.init();
        this._heroAvatar.updateUI(heroId, null, null, limit);
        this._textName.string = (name);
        this._textName.node.color = (Colors.getOfficialColor(officel));
        UIHelper.enableOutline(this._textName, Colors.getOfficialColorOutline(officel), 2);
        var [officalInfo, officalLevel] = G_UserData.getBase().getOfficialInfo(officel);
        if (officalLevel <= 0) {
            this._textOfficial.node.active = (false);
            return;
        }
        this._textOfficial.node.color = (Colors.getOfficialColor(officalLevel));
        UIHelper.enableOutline(this._textOfficial, officalLevel, 2)
        this._textOfficial.node.active = (true);
        this._textOfficial.string = ('[' + (officalInfo.name + ']'));
        this._heroAvatar.showTitle(titleId, "MineMoveAvatar");
    }
    setAction(action, loop) {
        this._heroAvatar.setAction(action, loop);
    }
    turnBack(needTurn) {
        this._heroAvatar.turnBack(needTurn);
    }

}